e2e/puppeteer-custom-environment.ts

```diff
@@ -20,26 +19,26 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
   // Take a screenshot right after failure
   async handleTestEvent(event, state) {
     switch (event.name) {
-    case 'test_fn_failure':
-      if (state.currentlyRunningTest.invocations > retryTimes) {
-        console.error(`Failed test:  "${event.test.name}"`);
-        const testName = fp.startCase(state.currentlyRunningTest.name).replace(/[^A-Z0-9]+/ig, '');
-        const screenshotDir = 'logs/screenshot';
-        await fs.ensureDir(screenshotDir);
+      case 'test_fn_failure':
+      case 'hook_failure':
+      case 'test_done':
+        if (state.currentlyRunningTest.invocations > retryTimes || event.test.errors.length > 0) {
```

Comment:
Try to cover all types of failures because screenshots were not happening in some cases of failures such as unhandled exceptions.